aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageUploadTest.java

```diff
@@ -0,0 +1,224 @@
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.hub.HubEvent;
+import com.amplifyframework.storage.StorageAccessLevel;
+import com.amplifyframework.storage.StorageException;
+import com.amplifyframework.storage.operation.StorageUploadFileOperation;
+import com.amplifyframework.storage.options.StorageUploadFileOptions;
+import com.amplifyframework.storage.result.StorageUploadFileResult;
+import com.amplifyframework.testutils.Await;
+
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferState;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Instrumentation test for operational work on upload.
+ */
+@SuppressWarnings("Indentation") // Doesn't seem to like lambda indentation
+public final class AWSS3StorageUploadTest extends StorageInstrumentationTestBase {
+
+    private static final StorageAccessLevel DEFAULT_ACCESS_LEVEL = StorageAccessLevel.PUBLIC;
+    private static final long DEFAULT_TIMEOUT_IN_SECONDS = 10;
+
+    private static final long LARGE_FILE_SIZE = 10 * 1024 * 1024L; // 10 MB
+    private static final long SMALL_FILE_SIZE = 100L;
+
+    private final String largeFileName = "large-test-" + System.currentTimeMillis();
+    private final String smallFileName = "small-test-" + System.currentTimeMillis();
+
+    private File largeFile;
+    private File smallFile;
+
+    private StorageUploadFileOptions options;
+    private String largeFileKey;
+    private String smallFileKey;
+
+    /**
+     * Sets up the required files for testing transfers.
+     * @throws Exception if fails to create temp files
+     */
+    @Before
+    public void setUp() throws Exception {
+        // Randomly write objects to upload
+        largeFile = createTempFile(largeFileName, LARGE_FILE_SIZE);
+        smallFile = createTempFile(smallFileName, SMALL_FILE_SIZE);
+
+        // Always interact with PUBLIC access for consistency
+        options = StorageUploadFileOptions.builder()
+                .accessLevel(DEFAULT_ACCESS_LEVEL)
+                .build();
+
+        largeFileKey = getS3Key(DEFAULT_ACCESS_LEVEL, largeFileName);
+        smallFileKey = getS3Key(DEFAULT_ACCESS_LEVEL, smallFileName);
+    }
+
+    /**
+     * Cleans up the S3 bucket for files that were uploaded
+     * during testing processes.
+     */
+    @After
+    public void cleanUp() {
```

Comment:
I just remembered that I may have set up a TTL on all objects in the bucket. Clean up may be best effort but could be test execution may be terminated for some reason. You can do the same for the bucket if this isn’t sharing the same resource as iOS

Reply:
Good idea!